python_modules/dagster/dagster/_core/definitions/auto_materialize_sensor_definition.py

```diff
@@ -1,12 +1,102 @@
 from typing import Any, Mapping, Optional, cast
 
+import pendulum
+
+import dagster._check as check
 from dagster._annotations import experimental
+from dagster._core.asset_graph_view.asset_graph_view import AssetGraphView, TemporalContext
 from dagster._core.definitions.asset_selection import CoercibleToAssetSelection
+from dagster._core.definitions.data_time import CachingDataTimeResolver
+from dagster._core.definitions.data_version import CachingStaleStatusResolver
+from dagster._core.definitions.declarative_scheduling.scheduling_condition_evaluator import (
+    SchedulingConditionEvaluator,
+)
+from dagster._core.definitions.run_request import SensorResult
+from dagster._utils.caching_instance_queryer import CachingInstanceQueryer
 
 from .asset_selection import AssetSelection
-from .sensor_definition import DefaultSensorStatus, SensorDefinition, SensorType
+from .sensor_definition import (
+    DefaultSensorStatus,
+    SensorDefinition,
+    SensorEvaluationContext,
+    SensorType,
+)
 from .utils import check_valid_name, normalize_tags
 
+EXECUTABLE_AMP_SENSOR_DESC = "Experimental user space scheduling condition evaluation sensor."
+
+
+def evaluate_scheduling_conditions(context: SensorEvaluationContext):
+    from dagster._core.definitions.asset_daemon_context import build_run_requests
+    from dagster._daemon.asset_daemon import (
+        asset_daemon_cursor_from_instigator_serialized_cursor,
+        asset_daemon_cursor_to_instigator_serialized_cursor,
+    )
+
+    asset_graph = check.not_none(context.repository_def).asset_graph
+
+    instance_queryer = CachingInstanceQueryer(
```

Comment:
It would be nice to know what storage calls this will unleash if we were to ship it today / how we feel about the runtime of those storage calls when they are run over graphql at scale and we no longer have the ability to easily change the API boundary anymore. (We know that there are some that are clearly not at the correct boundary right now, like the asset status cache stuff)

Reply:
I don't think we are going to be able to answer that in the course of this PR. However I do think we can do a lot of obvious things in follow up such as fetching the `AssetRecord` all assets covered by the sensor as similar.